tests/base/src/test/java/org/keycloak/tests/admin/authentication/AbstractAuthenticationTest.java

```diff
@@ -0,0 +1,234 @@
+        flow.setTopLevel(topLevel);
+        flow.setBuiltIn(builtIn);
+        return flow;
+    }
+
+    AuthenticationExecutionInfoRepresentation newExecInfo(String displayName, String providerId, Boolean configurable,
+                                                          int level, int index, String requirement, Boolean authFlow, String[] choices,
+                                                          int priority) {
+
+        AuthenticationExecutionInfoRepresentation execution = new AuthenticationExecutionInfoRepresentation();
+        execution.setRequirement(requirement);
+        execution.setDisplayName(displayName);
+        execution.setConfigurable(configurable);
+        execution.setProviderId(providerId);
+        execution.setLevel(level);
+        execution.setIndex(index);
+        execution.setAuthenticationFlow(authFlow);
+        execution.setPriority(priority);
+        if (choices != null) {
+            execution.setRequirementChoices(Arrays.asList(choices));
+        }
+        return execution;
+    }
+
+    void addExecInfo(List<AuthenticationExecutionInfoRepresentation> target, String displayName, String providerId, Boolean configurable,
+                 int level, int index, String requirement, Boolean authFlow, String[] choices, int priority) {
+
+        AuthenticationExecutionInfoRepresentation exec = newExecInfo(displayName, providerId, configurable, level, index, requirement, authFlow, choices, priority);
+        target.add(exec);
+    }
+
+    AuthenticatorConfigRepresentation newConfig(String alias, String[] keyvalues) {
+        AuthenticatorConfigRepresentation config = new AuthenticatorConfigRepresentation();
+        config.setAlias(alias);
+
+        if (keyvalues == null) {
+            throw new IllegalArgumentException("keyvalues == null");
+        }
+        if (keyvalues.length % 2 != 0) {
+            throw new IllegalArgumentException("keyvalues should have even number of elements");
+        }
+
+        LinkedHashMap<String, String> params = new LinkedHashMap<>();
+        for (int i = 0; i < keyvalues.length; i += 2) {
+            params.put(keyvalues[i], keyvalues[i + 1]);
+        }
+        config.setConfig(params);
+        return config;
+    }
+
+    String createFlow(AuthenticationFlowRepresentation flowRep) {
+        return createFlow(flowRep, true);
+    }
+
+    String createFlow(AuthenticationFlowRepresentation flowRep, boolean autoDelete) {
+        Response response = authMgmtResource.createFlow(flowRep);
+        Assertions.assertEquals(201, response.getStatus());
+        response.close();
+        String flowId = ApiUtil.getCreatedId(response);
+        if (autoDelete) {
+            managedRealm.cleanup().add(r -> r.flows().deleteFlow(flowId));
+        }
+//        assertAdminEvents.assertEvent(testRealmId, OperationType.CREATE, AssertAdminEvents.isExpectedPrefixFollowedByUuid(AdminEventPaths.authFlowsPath()), flowRep, ResourceType.AUTH_FLOW);
```

Comment:
trailing comment can probably be deleted

Reply:
Removing